### DISCUSSION TITLE
identifier is private IP. Not global IP

### DISCUSSION THREAD

**Discussion by maezonoyusaku** [strongswan/strongswan#1396] [Q&A] [answered]
The administrator of the destination vpn router told me that "peer ID is set to the private IP of your router(strongswan). The ID should be set to the global IP". Can you tell me how to set the ID in strongswan?

**Comment by strongX509** [comment, marked as answer]
In your **`xyz`** connection definition in **`swanctl.conf`** add a **`local`** section:
````
connections {
  xyz {
    local {
      id = <global IP>
    }
    ...
  }
}
````
This is only required if due to a NAT router the internal IP is not the same as the external IP, since the default for **`local.id`** is the [internal] IP address of the outgoing network interface.

**Reply by maezonoyusaku** [reply]
Thank you. I'll try it now.

**Reply by maezonoyusaku** [reply]
I use ipsec.conf, but I didn't know swanctl.conf. I'm trying to figure out how to use swanctl.conf.

**Reply by maezonoyusaku** [reply]
I found a correspondence table. https://wiki.strongswan.org/projects/strongswan/wiki/Fromipsecconf

**Reply by maezonoyusaku** [reply]
I restarted strongswan with the following changes in /etc/ipsec.conf. I'll ask the administrator on the other side.

```
conn xxxx
        aaa_identity=xxx(my global IP)
        ...
        ...
```

**Reply by maezonoyusaku** [reply]
The following may be correct..? (I'm checking now)

```
conn xxxx
        leftid==xxx(my global IP)
        ...
        ...
```

**Reply by maezonoyusaku** [reply]
The administrator on the other side seems to have gone home, but the results of the ipsec command look fine. Thank you!

```
# ipsec statusall
...
Security Associations (1 up, 0 connecting):
...
```
